pkg/operator/v1helpers/helpers.go

```diff
@@ -347,3 +349,65 @@ func InjectObservedProxyIntoContainers(podSpec *corev1.PodSpec, containerNames [
 
 	return nil
 }
+
+func IsDeploymentProgressing(conditionName string, deployment *appsv1.Deployment, status *operatorv1.OperatorStatus) (bool, string) {
+	if deployment.Generation != deployment.Status.ObservedGeneration {
+		return true, "Waiting for Deployment to act on changes"
+	}
+
+	var expectedReplicas int32
+	if deployment.Spec.Replicas != nil {
+		expectedReplicas = *deployment.Spec.Replicas
+	}
+
+	if expectedReplicas > deployment.Status.UpdatedReplicas {
+		// Verify if the current deployment generation is the same as the one that this controller
+		// dealt with in the previous sync. If they are not the same, we're definitely rolling
+		// out new config, so we are progressing.
+		lastGeneration := resourcemerge.ExpectedDeploymentGeneration(deployment, status.Generations)
+		if deployment.Generation != lastGeneration {
+			return true, "Waiting for Deployment to deploy node pods"
+		}
+
+		// At this point we know two things: we already synced the current generation of the deployment,
+		// but the Deployment Controller is still rolling out pods. However, we don't know whether those
```

Comment:
if the deployment controller is still rolling out pods, why would we want to avoid being progressing?  How often are you scaling up?

Reply:
[The bug points out][1] that [the API docs for `Progressing` say][2]:

> Progressing indicates that the operator is actively rolling out new code, propagating config changes, or otherwise moving from one steady state to another.  Operators should not report progressing when they are reconciling a previously known state.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1940286#c0
[2]: https://github.com/openshift/api/blob/8356aa4d4afb94790d3ad58c4debe0e1bdabcbe9/config/v1/types_cluster_operator.go#L147-L151